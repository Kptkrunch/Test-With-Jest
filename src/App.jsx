import React, { Component } from 'react';
import { Button } from 'react-bootstrap';
import Gift from './components/Gift.jsx';

class App extends Component {
    constructor() {
        super();
        this.state = {
            gifts: []
        }
        this.addGift = this.addGift.bind(this);
        this.removeGift = this.removeGift.bind(this);
    }

    addGift() {
        const { gifts } = this.state;
        const ids = this.state.gifts.map( gift => gift.id );
        const max_id = ids.length > 0 ? Math.max(...ids) : 0;
        gifts.push({ id: max_id + 1 })
        this.setState({ gifts })
    }

    removeGift(id) {
        const gifts = this.state.gifts.filter( gift => gift.id != id);
        this.setState({ gifts: gifts });
    }

    render() {
        return (
            <div>
                <h1>Gift Giver</h1>
                <div className="gift-list">
                    {
                        this.state.gifts.map((gift, i) => {
                            return (
                                <Gift 
                                    key={i} 
                                    gift={gift}
                                    removeGift={this.removeGift}
                                />
                            )
                        })
                    }
                </div>
                <Button 
                    className="btn-add" 
                    onClick={this.addGift}
                >
                Add Gift
                </Button> 
            </div>
        )
    }
}

export default App;

