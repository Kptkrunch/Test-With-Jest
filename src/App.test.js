import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('App', () => {
    it('Should correctly render App in debug mode', () => {
        const app = shallow(<App debug />);

        expect(app).toMatchSnapshot();
    });
});