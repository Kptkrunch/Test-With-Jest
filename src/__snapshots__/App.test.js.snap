// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should correctly render App in debug mode 1`] = `
<div>
  <h1>
    Hello World
  </h1>
</div>
`;
